mixer/template/tracespan/tracespan.proto

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package traceSpan;
+
+import "google/protobuf/timestamp.proto";
+import "mixer/v1/config/descriptor/value_type.proto";
+import "mixer/v1/template/extensions.proto";
+
+option (istio.mixer.v1.template.template_variety) = TEMPLATE_VARIETY_REPORT;
+
+// TraceSpan represents an individual span within a distributed trace.
+//
+// When writing the configuration, the value for the fields associated with this template can either be a
+// literal or an [expression](https://istio.io/docs/reference/config/mixer/expression-language.html). Please note that if the datatype of a field is not istio.mixer.v1.config.descriptor.ValueType,
+// then the expression's [inferred type](https://istio.io/docs/reference/config/mixer/expression-language.html#type-checking) must match the datatype of the field.
+//
+// Example config:
+// ```
+// apiVersion: "config.istio.io/v1alpha2"
+// kind: tracespan
+// metadata:
+//   name: default
+//   namespace: istio-system
+// spec:
+//   traceId: request.headers["x-b3-traceid"]
+//   spanId: request.headers["x-b3-spanid"] | ""
```

Comment:
are the x-b3-spanid and  x-b3-parentspanid well known header names. If we expect operators to use these for most case, consider explaining a bit about whey those tags mean.

Reply:
these are well-known header names (from the zipkin set of headers).  they are described in full on our distributed tracing task. I've added a link to that Task here.

Reply:
Cool, that link will be helpful.